src/Context/GlobalContext/StateReducer.js

```diff
@@ -47,6 +50,35 @@ export const stateReducerFun = (state, action) => {
                     categoryName: [...state.filters.categoryName, action.payload],
                 },
             };
+
+            case STAR_RATING: 
+            
+            if(action.payload === 4){
+                return{...state, filters: {...state.filters, rating: action.payload}}
+            }
+            if(action.payload === 3){
+                return{...state, filters: {...state.filters, rating: action.payload}}
+            }
+            if(action.payload === 2){
+                return{...state, filters: {...state.filters, rating: action.payload}}
+            }
+            if(action.payload === 1){
+                return{...state, filters: {...state.filters, rating: action.payload}}
+            }
+
+            case PRICE_RANGE:
+            
+            return {...state, filters: {...state.filters, priceRange: action.payload}}
+
+            case CLEAR_FILTER:
+
+            return {
+               ...state, filters: {...state.filters,  sortBy: "",
+               priceRange: 500,
```

Comment:
Suggestion: we can create a method which returns the max price among all the products. That way we don't need to worry about the range of price.

Reply:
Done